### ISSUE TITLE
[Feature Request]: Naming options for "Save image" function

### ISSUE THREAD

**Issue by I2ujbNGakP8KfLKbho0Z** [ollm/OpenComic#286]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/ollm/OpenComic/issues) for a feature request that matches the one I want to file, without success.

### Describe your suggested feature

Currently, when using the "Save image" function, the file name gets set as the original file name of the page that gets saved. It'd be nice to be able to override that. One example would be wanting the file name of the saved page to be: 
`[name of folder the file you are reading is in] - [name of the file you are reading] - [current page of the file]`

That'd be good to immediately be able to recognize where a saved image comes from and to prevent duplicate name conflicts in the case the page file names of two pages that are saved are the same.

**Comment by dajotim937** [comment]
Agreed. Instead of selecting folder when saving single image, it would be better to change save window to "save file" with entering filename.
Also, I think, default filename for image from cbz file should be "<cbz file name> - <name of image>".

And also it would fix current bug: if file with same name exists in folder, OpenComic doesn't save anything. Which is good, because you never know what it could override without any alert messages, but popup message in OpenComic says that "image saved".

**Comment by ollm** [comment]
I changed the folder selection dialog to the file save dialog 4b01440b70b52e10c3cbcb37f1379081242c523f, with this you can now change the file name, also it is possible to use some vars to get data from the original file, as follows.

Default filename: `[parentFolder] - [folder] - [image] - [page]`

- `[page]` The page number, leading zeros are added if necessary.
- `[pageInt]`  The page number without leading zeros.
- `[image]` Original image name without extension.
- `[folder]` The folder name or file name without extension if is a cbz, cbr, etc.
- `[parentFolder]` The parent folder name.

You can try it in the following build: https://mega.nz/file/bXRFFQYb#M6MEKO-EY0W5BebAAeTFH4lsgFwBbcRoW4WlJ6sK1I8

**Comment by dajotim937** [comment]
Maybe setting filename template from settings, because for now template reset to default for next save and also there is no "replace" warning if you save page with same name. I assume because filename template doesn't collide before actual values at saving. And OpenComic shows that file saved successfuly with the same name from template, but it doesn't save anything, because file with that name already exists.

1. Save page.
2. Copy full name of saved page
3. Delete file
4. Rename different image to copied name
5. Save same page again
6. "Saved successfuly"
7. Different image with saved page name is exists (as it should, because it shouldn't be replaced without warning).
Result: No warning, wrong pop-up "saved successfuly".

**Comment by ollm** [comment]
Closed as completed in 4b01440b70b52e10c3cbcb37f1379081242c523f, 7def9eca58ceb24fb3444a3daf3ed81178a13fd1, 05c9b315788fec63aee2333b0bfb0588cfcf536a, 47e8de7b73b1ec354000c1455ff651039abd49e1 and available in [v1.4.0](https://github.com/ollm/OpenComic/releases)

Now you can set a template for the file name from the app settings, also if the name already exists, the images will be saved with an incremental number, for example: `image (1).jpg`, `image (2).jpg` , etc. Also, you can set up a folder to automatically save images to.

![Image](https://github.com/user-attachments/assets/c1bab856-4a5b-433e-b977-237d597d58ee)
